// near-react-hooks/esm/NearSignin.js
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from 'react';
import { useNearWallet } from './hooks';
var DEFAULT_STYLE = {
    padding: '0.8rem 1.5rem',
    border: 'none',
    borderRadius: '30px',
    backgroundColor: 'rgb(0, 114, 206)',
    color: 'white',
    cursor: 'pointer'
};
/**
 *
 * `NearSignin` renders a button used to signin to a NEAR account.
 */
export var NearSignin = function (_a) {
    var _b = _a.text, text = _b === void 0 ? 'Signin with NEAR' : _b, className = _a.className, style = _a.style, rest = __rest(_a, ["text", "className", "style"]);
    var wallet = useNearWallet();
    if (wallet.isSignedIn()) {
        return (React.createElement("p", { className: className, style: className !== undefined ? null : style !== undefined ? style : DEFAULT_STYLE }, wallet.getAccountId()));
    }
    else {
        return (React.createElement("button", { className: className, style: className !== undefined ? null : style !== undefined ? style : DEFAULT_STYLE, onClick: function () { return wallet.requestSignIn(rest); } }, text));
    }
};
